### PR TITLE
fix back button for root Files view on mobile

### PR DESCRIPTION
**Pull request by songgao**
The bug was that, on mobile, the `folderRoute` is not a direct children of the SettingsTab anymore. Instead, there's a real `fsRoute` now, whose default selected route is `folderRoute`. So one `navigateUp()` doesn't work here. However, as we want to use `putActionIfOnPath`, it's not easy to put `navigateUp()` twice. So instead just add a `navigateUp2` action.

### CODE REVIEW

**Comment by chrisnojima** [comment]
Let's not do either of these. Either nav directly where you want to go (navigateTo) or change you're routing hierarchy

**Comment by songgao** [comment]
OK; also I just realized this PR has a new bug where it doesn't do it properly for non-/keybase paths. Will push new commits soon to use `navigateTo`

File: shared/fs/header/container.js [hunk 1]
```diff
@@ -22,7 +25,10 @@ const mapDispatchToProps = (dispatch, {path, routePath}: OwnProps) => ({
         ...Util.tlfToParticipantsOrTeamname(Types.pathToString(path)),
       })
     ),
-  onBack: () => dispatch(putActionIfOnPath(routePath, navigateUp())),
+  onBack:
+    isMobile && path === Constants.defaultPath
+      ? () => dispatch(putActionIfOnPath(routePath, navigateTo([TabsConstants.settingsTab])))
```
Comment (chrisnojima) [inline shared/fs/header/container.js]: why not just fix the route so navigateUp works

Comment (songgao) [reply shared/fs/header/container.js]: I could figure out a way to make switch to work when user taps “Cancel” in the destination picker (for moveOrCopy), so just went with having folders route and destination picker route as siblings, which means they are both Uber fsRoute and that the fsRoute itself doesn’t render a component. See notes in https://github.com/keybase/client/blob/master/shared/fs/routes.js

Please let me know if there are better route fixes!

Comment (songgao) [reply shared/fs/header/container.js]: oops s/could/couldn't

**Comment by songgao** [comment]
Talked on Meet with Nojima; we are gonna change the route design so that DestinationPicker and Folder view share the same routes. Will close this now and open a new one when it's ready.

**Pull request closed without merging**